File: schedule-server/src/main/java/com/aspire/schedule/repository/entity/Task.java
package com.aspire.schedule.repository.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("task")
public class Task {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long goalId;
    private Long userId;
    private String title;
    private String description;
    private Integer priority;
    private String status;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private BigDecimal estimatedHours;
    private Long parentTaskId;
    private String dependencyType;
    private String sourceType;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: schedule-web/src/types/task.ts
export interface Task {
  id: number;
  goalId: number;
  userId: number;
  title: string;
  description: string;
  priority: 1 | 2 | 3;
  status: 'TODO' | 'IN_PROGRESS' | 'DONE' | 'DELAYED' | 'CANCELLED';
  startTime: string;
  endTime: string;
  estimatedHours: number;
  parentTaskId: number;
  dependencyType: string;
  sourceType: 'MANUAL' | 'AI_GENERATED';
  createdAt: string;
  updatedAt: string;
}

export interface TaskUpdateDTO {
  title?: string;
  description?: string;
  priority?: number;
  startTime?: string;
  endTime?: string;
}

export interface TaskStatusDTO {
  status: string;
}
